### DISCUSSION TITLE
Renovate takes too much time and too much resources

### DISCUSSION THREAD

**Discussion by KnisterPeter** [renovatebot/renovate#19906] [Request Help] [answered]
Hi there,

we have a monorepo (closed source in GitHub Enterprise) which is quite big.
If we run renovate (custom integration with a GitHub App) on that repo it took more than 20 minutes to complete. While that it takes 3+GB RAM. The dependency extraction shows 9000+ entries.

We already have the presets `:prHourlyLimit2` and `:prConcurrentLimit10` enabled. Unfortunately this does not limit the resources and runtime.

Is there a way to configure a 'partial' renovation run of a repository? (Hope this questions makes sense).

**Comment by rarkins** [comment, marked as answer]
I don't have much interest in the RAM question but feel free to contribute a PR if you find some low hanging fruit.

Renovate at debug log level outputs statistics at the end of a run. You should first look at that and work out where the time is going.

You should consider enabling `repositoryCache` as it can speed things up quite a lot, and we continue to work on it for further gains.

**Reply by KnisterPeter** [reply]
Thanks, that are good first pointer where I can try to improve the situation.
I'll mark this as answered and will come back with a more detailed question if needed.
